Unit test fails at `clock.test.ts` from time to time

### Describe the bug

Sometimes `clock.test.ts` fails. Re-running it usually makes the error go away. Same behaviour is observed when running `clock.test.ts` on my local machine.

```
@Chainsafe validator % yarn vitest test/unit/utils/clock.test.ts 
yarn run v1.22.22
$ node_modules/.bin/vitest test/unit/utils/clock.test.ts

 DEV  v1.4.0 lodestar/packages/validator

 ❯ test/unit/utils/clock.test.ts (6)
   ❯ util / Clock (6)
     ✓ Should call on slot
     ✓ Should stop calling on slot after stop()
     ✓ Should call on epoch
     ❯ getCurrentSlot (3)
       × 'should return next slot after 11.5s'
       ✓ 'should return next slot after 12s'
       ✓ 'should return next slot after 12.5s'

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  test/unit/utils/clock.test.ts > util / Clock > getCurrentSlot > 'should return next slot after 11.5s'
AssertionError: expected 8780981 to be 8780982 // Object.is equality

- Expected
+ Received

- 8780982
+ 8780981

 ❯ test/unit/utils/clock.test.ts:96:61
     94|       const currentSlot = getCurrentSlotAround(testConfig, genesisTime);
     95|       vi.advanceTimersByTime(delta * 1000);
     96|       expect(getCurrentSlotAround(testConfig, genesisTime)).toBe(currentSlot + 1);
       |                                                             ^
     97|     });
     98|   });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

 Test Files  1 failed (1)
      Tests  1 failed | 5 passed (6)
   Start at  21:56:05
   Duration  731ms (transform 435ms, setup 12ms, collect 591ms, tests 6ms, environment 0ms, prepare 33ms)


 FAIL  Tests failed. Watching for file changes...
       press h to show help, press q to quit
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
@Chainsafe validator % 
@Chainsafe validator % 
@Chainsafe validator % yarn vitest test/unit/utils/clock.test.ts
yarn run v1.22.22
$ lodestar/node_modules/.bin/vitest test/unit/utils/clock.test.ts

 DEV  v1.4.0 lodestar/packages/validator

 ✓ test/unit/utils/clock.test.ts (6)
   ✓ util / Clock (6)
     ✓ Should call on slot
     ✓ Should stop calling on slot after stop()
     ✓ Should call on epoch
     ✓ getCurrentSlot (3)
       ✓ 'should return next slot after 11.5s'
       ✓ 'should return next slot after 12s'
       ✓ 'should return next slot after 12.5s'

 Test Files  1 passed (1)
      Tests  6 passed (6)
   Start at  21:56:12
   Duration  560ms (transform 354ms, setup 12ms, collect 446ms, tests 4ms, environment 0ms, prepare 23ms)


 PASS  Waiting for file changes...
       press h to show help, press q to quit
✨  Done in 3.84s.

```

Workflows log [here](https://github.com/ChainSafe/lodestar/actions/runs/8949227035/job/24583334814?pr=6715)

### Expected behavior

The test should pass every time

### Steps to reproduce

Run `yarn vitest test/unit/utils/clock.test.ts` multiple times

### Additional context

_No response_

### Operating system

Linux

### Lodestar version or commit hash

v.1.18.0

--- a/packages/validator/test/unit/utils/clock.test.ts
+++ b/packages/validator/test/unit/utils/clock.test.ts
@@ -84,6 +84,14 @@ describe("util / Clock", function () {
     // eslint-disable-next-line @typescript-eslint/naming-convention
     const testConfig = {SECONDS_PER_SLOT: 12} as BeaconConfig;
     const genesisTime = Math.floor(new Date("2021-01-01").getTime() / 1000);
+
+    // Tests can fail under certain time slots, overriding the system time 
+    // with a specific value allows us to run tests deterministically
+    // See : https://github.com/ChainSafe/lodestar/issues/6734
+    beforeEach(() => {
+      vi.setSystemTime(genesisTime * 1000);
+    });
+
     const testCase: {name: string; delta: number}[] = [
       {name: "should return next slot after 11.5s", delta: 11.5},
       {name: "should return next slot after 12s", delta: 12},

--- a/packages/validator/test/unit/utils/clock.test.ts
+++ b/packages/validator/test/unit/utils/clock.test.ts
@@ -85,9 +85,8 @@ describe("util / Clock", function () {
     const testConfig = {SECONDS_PER_SLOT: 12} as BeaconConfig;
     const genesisTime = Math.floor(new Date("2021-01-01").getTime() / 1000);
 
-    // Tests can fail under certain time slots, overriding the system time 
+    // Tests can fail under certain time slots, overriding the system time
     // with a specific value allows us to run tests deterministically
-    // See : https://github.com/ChainSafe/lodestar/issues/6734
     beforeEach(() => {
       vi.setSystemTime(genesisTime * 1000);
     });

